### PR TITLE
[TASK] Update unit tests for TYPO3 master

### PR DESCRIPTION
**Pull request by NamelessCoder**

### CODE REVIEW

**Comment by NamelessCoder** [comment]
Very WIP - just demonstrating odd failures.

File: Classes/View/UncacheTemplateView.php [hunk 1]
```diff
@@ -22,6 +24,29 @@
...
+    protected $templateParser;
+
+    /**
+     * @var TemplateCompiler|\TYPO3Fluid\Fluid\Core\Compiler\TemplateCompiler
+     */
+    protected $templateCompiler;
+
+    /**
+     * @param ObjectManagerInterface $objectManager
+     * @return vpoid
```
Comment (cedricziel) [inline Classes/View/UncacheTemplateView.php]: `void`

File: composer.json [hunk 2]
```diff
@@ -23,10 +23,13 @@
         }
     },
     "require": {
-        "php": ">=5.5.0"
+        "php": ">=5.5.0",
+        "psr/log": "1.0.0 as 1.0.2",
+        "mso/idna-convert": "1.1.0 as 0.9.1",
+        "typo3/cms": "dev-master as 7.6.11"
```
Comment (cedricziel) [inline composer.json]: Are you sure about this one?

Comment (NamelessCoder) [inline composer.json]: For the WIP phase, yes ;)

**Comment by NamelessCoder** [comment]
Alright - these changes should make VHS run and run tests on TYPO3 dev-master (and therefore 8.4) but the PR still needs to be thoroughly cleaned up and perhaps divided into multiple ones. Also, Travis build will require some way to run TYPO3 dev-master and latest release, with failures allowed for dev-master, and only when php version is >= 7.0.

Still lots to do but a step in the right direction :)

**Comment by NamelessCoder** [comment]
@cedricziel if you're okay with this then we may have to break the build for a brief moment between merging this patch and merging https://github.com/FluidTYPO3/vhs/pull/1176. Neither will work without the other one, and it'd make sense to merge this one first ;)

**Comment by cedricziel** [comment]
I see you're still updating your patchset. Ping me when you're ready.

**Comment by cedricziel** [comment]
Oh by the way: To set paths correctly, you might want to consider a similar structure as https://github.com/georgringer/news/blob/master/composer.json#L59-L70 to trick the installer scripts

**Comment by NamelessCoder** [comment]
Any idea why the 7.6 test matrixes do not create the symlinks?

**Comment by NamelessCoder** [comment]
Strike that; it creates the links but cannot locate the typo3 dir, even when I specify TYPO3_PATH__WEB...

**Comment by NamelessCoder** [comment]
Finally! Now the tests run the same on both versions but of course fail with the "trait not found" message. If we merge this and I then rebase https://github.com/FluidTYPO3/vhs/pull/1176 the build should become green again.

Anyone awake for merging? :)

File: Classes/ViewHelpers/DebugViewHelper.php [hunk 3]
```diff
@@ -165,10 +165,10 @@ public function render()
     public function setChildNodes(array $childNodes)
     {
         foreach ($childNodes as $childNode) {
-            if (true === $childNode instanceof ViewHelperNode) {
+            if (true === $childNode instanceof ViewHelperNode || $childNode instanceof \TYPO3Fluid\Fluid\Core\Parser\SyntaxTree\ViewHelperNode) {
```
Comment (cedricziel) [inline Classes/ViewHelpers/DebugViewHelper.php]: Sidenote: I prefer aliased imports such as `StandaloneFluidViewHelperNode` or `LegacyViewHelperNode`

File: Tests/Unit/ViewHelpers/Condition/Page/IsChildPageViewHelperTest.php [hunk 4]
```diff
@@ -9,7 +9,7 @@
  */
 
 use FluidTYPO3\Vhs\Tests\Unit\ViewHelpers\AbstractViewHelperTest;
-use TYPO3\CMS\Core\Database\DatabaseConnection;
+use TYPO3\CMS\Core\Database\ConnectionPool;
```
Comment (cedricziel) [inline Tests/Unit/ViewHelpers/Condition/Page/IsChildPageViewHelperTest.php]: Unused import

Comment (NamelessCoder) [inline Tests/Unit/ViewHelpers/Condition/Page/IsChildPageViewHelperTest.php]: I'll do all of the unused imports in another PR - there are quite a few of them in various classes!

File: Classes/ViewHelpers/DebugViewHelper.php (hunk 3 above)
Comment (NamelessCoder) [inline Classes/ViewHelpers/DebugViewHelper.php]: Coming right up.

**Comment by cedricziel** [comment]
Okay. Next.

**Pull request merged**